src/api/js/src/high-level/types.ts

```diff
@@ -0,0 +1,520 @@
+  readonly __typename: 'Model';
+
+  readonly ctx: Context<Name>;
+  readonly ptr: Z3_model;
+
+  get length(): number;
+
+  entries(): IterableIterator<[number, FuncDeclRef<Name>]>;
+  keys(): IterableIterator<number>;
+  values(): IterableIterator<FuncDeclRef<Name>>;
+  decls(): FuncDeclRef<Name>[];
+  sexpr(): string;
+  eval(expr: BoolRef<Name>, modelCompletion?: boolean): BoolRef<Name>;
+  eval(expr: ArithRef<Name>, modelCompletion?: boolean): ArithRef<Name>;
+  eval(expr: ExprRef<Name>, modelCompletion?: boolean): ExprRef<Name>;
+  get(i: number): FuncDeclRef<Name>;
+  get(from: number, to: number): FuncDeclRef[];
+  get(declaration: FuncDeclRef<Name>): FuncInterp<Name> | ExprRef<Name>;
+  get(constant: ExprRef<Name>): ExprRef<Name>;
+  get(sort: SortRef<Name>): AstVector<AnyExpr<Name>, Name>;
+}
+
+export interface SortRef<Name extends string = any> extends AstRef<Name, Z3_sort> {
+  readonly __typename: 'SortRef' | BoolSortRef['__typename'] | ArithSortRef['__typename'] | BitVecSortRef['__typename'];
+
+  kind(): Z3_sort_kind;
+  subsort(other: SortRef<Name>): boolean;
+  cast(expr: CoercibleToExpr<Name>): ExprRef<Name>;
+  name(): string | number;
+  eqIdentity(other: SortRef<Name>): boolean;
+  neqIdentity(other: SortRef<Name>): boolean;
+}
+
+export interface FuncInterp<Name extends string = any> {
+  readonly __typename: 'FuncInterp';
+
+  readonly ctx: Context<Name>;
+  readonly ptr: Z3_func_interp;
+}
+
+export interface FuncDeclRef<Name extends string = any> extends AstRef<Name, Z3_func_decl> {
+  readonly __typename: 'FuncDeclRef';
+
+  name(): string | number;
+  arity(): number;
+  domain(i: number): SortRef<Name>;
+  range(): SortRef<Name>;
+  kind(): Z3_decl_kind;
+  params(): (number | string | Z3_symbol | SortRef<Name> | ExprRef<Name> | FuncDeclRef<Name>)[];
+  call(...args: ExprRef<Name>[]): AnyExpr<Name>;
+}
+
+export interface ExprRef<Name extends string = any, Sort extends SortRef<Name> = AnySort<Name>, Ptr = unknown>
+  extends AstRef<Name, Ptr> {
+  readonly __typename:
+    | 'ExprRef'
+    | BoolRef['__typename']
+    | PatternRef['__typename']
+    | QuantifierRef['__typename']
+    | LambdaRef['__typename']
+    | ArithRef['__typename']
+    | BitVecRef['__typename'];
+
+  sort(): Sort;
+  eq(other: CoercibleToExpr<Name>): BoolRef<Name>;
+  neq(other: CoercibleToExpr<Name>): BoolRef<Name>;
+  params(): ReturnType<FuncDeclRef<Name>['params']>;
+  decl(): FuncDeclRef<Name>;
+  numArgs(): number;
+  arg(i: number): AnyExpr<Name>;
+  children(): AnyExpr<Name>[];
+}
+
+export interface BoolSortRef<Name extends string = any> extends SortRef<Name> {
+  readonly __typename: 'BoolSortRef';
+
+  cast(expr: BoolRef<Name> | boolean): BoolRef<Name>;
+  cast(expr: CoercibleToExpr<Name>): never;
+}
+
+export interface BoolRef<Name extends string = any> extends ExprRef<Name, BoolSortRef<Name>, Z3_ast> {
+  readonly __typename: 'BoolRef';
+
+  mul(other: BoolRef | boolean): BoolRef<Name>;
```

Comment:
It looks like this class wasn't finished?

Also this is missing the type parameter on the argument type.